v2/kafka-to-bigquery/src/main/java/com/google/cloud/teleport/v2/templates/KafkaToBigQueryFlex.java

```diff
@@ -248,27 +252,74 @@ public static PipelineResult runAvroPipeline(
-        writeResult = kafkaRecords.apply(AvroTransform.of(options));
-      }
-
-      if (options.getSchemaRegistryConnectionUrl() != null && options.getOutputDataset() == null) {
+      if (options.getSchemaFormat().equals("SINGLE_SCHEMA_FILE")) {
+
+        if (options.getConfluentAvroSchemaPath() != null && options.getOutputTableSpec() == null) {
+          throw new IllegalArgumentException("An output BigQuery table is required.");
+        }
+
+        if (options.getConfluentAvroSchemaPath() != null && options.getOutputTableSpec() != null) {
+          writeResult =
+              kafkaRecords.apply(
+                  AvroTransform.of(
+                      options.getReadBootstrapServerAndTopic().split(":")[2],
+                      options.getMessageFormat(),
+                      options.getBinaryAvroSchemaPath(),
+                      options.getConfluentAvroSchemaPath(),
+                      options.getOutputTableSpec(),
+                      options.getWriteDisposition(),
+                      options.getCreateDisposition(),
+                      options.getNumStorageWriteApiStreams(),
+                      options.getStorageWriteApiTriggeringFrequencySec(),
+                      options.getPersistKafkaKey(),
+                      options.getUseAutoSharding()));
+        }
+      } else if (options.getSchemaFormat().equals("SCHEMA_REGISTRY")) {
+
+        if (options.getSchemaRegistryConnectionUrl() != null
+            && options.getOutputDataset() == null) {
+          throw new IllegalArgumentException(
+              "An output BigQuery dataset is required. It will be used to create tables per schema.");
+        }
+
+        if (options.getSchemaRegistryConnectionUrl() != null
+            && options.getOutputDataset() != null) {
+          writeResult =
+              kafkaRecords.apply(
+                  AvroDynamicTransform.of(
+                      options.getSchemaRegistryConnectionUrl(),
+                      options.getOutputProject(),
+                      options.getOutputDataset(),
+                      options.getBqTableNamePrefix(),
+                      options.getWriteDisposition(),
+                      options.getCreateDisposition(),
+                      options.getNumStorageWriteApiStreams(),
+                      options.getStorageWriteApiTriggeringFrequencySec(),
+                      options.getPersistKafkaKey(),
+                      options.getUseAutoSharding()));
+        }
+      } else {
         throw new IllegalArgumentException(
-            "An output BigQuery dataset is required. It will be used to create tables per schema.");
-      }
-
-      if (options.getSchemaRegistryConnectionUrl() != null && options.getOutputDataset() != null) {
-        writeResult = kafkaRecords.apply(AvroDynamicTransform.of(options));
+            "There is a parameter misconfiguration, please set Schema Format parameter.");
```

Comment:
I'd add more details to the error message, this message can be confusing if the schemaFormat param is set to something, but that value is not supported.

```
throw new IllegalArgumentException("Unsupported schemaFormat parameter value: " + options.getSchemaFormat())
```

Reply:
done